shard_consumer.go

```diff
@@ -188,7 +189,11 @@ mainloop:
 
 		if err != nil {
 			if awsErr, ok := err.(awserr.Error); ok {
-				k.config.logger.Log("Got error: %s (%s) retry count is %d / %d", awsErr.Message(), awsErr.OrigErr(), retryCount, maxErrorRetries)
+				origErrStr := ""
```

Comment:
Would replacing the code with `k.config.logger.Log("Got error: %s (%s) retry count is %d / %d", awsErr.Message(), aws.StringValue(awsErr.OrigErr()), retryCount, maxErrorRetries)` do the same thing, and if so thought son which is more clear?

Reply:
That would be nicer than what is currently there. It's not quite the same as my approach. The downside of your approach is the log output would have unnecessary `()` in it when `awsErr.OrigErr() == nil`. The upside is the code is simpler.

I suppose it depends on whether you would prefer to optimise for nicer log lines or nicer code. I personally would prefer the former, but I don't feel strongly about this. Happy to incorporate your suggestion if you would like?

Reply:
Fair enough.